### PR TITLE
feat: add job management for subclient cerberus

### PR DESCRIPTION
**Pull request by azrod**
This pull request introduces significant updates to the `cav` package, focusing on enhancing job handling for the `Cerberus` and `VMware` subclients, adding middleware functionality, and improving testing coverage. Key changes include the addition of request middleware support, updates to job parsing logic, and the implementation of new unit tests for better validation and error handling.

### Enhancements to job handling:

* [`cav/subclient_cerberus_jobs.go`](diffhunk://#diff-61d4b4c7d5886db749682b4648856df3ed30381b5b71d4d0291ef54e91a44aeaR1-R197): Added support for parsing Cerberus job creation responses and introduced a new endpoint `JobCerberus` for handling job-related operations. Includes logic for refreshing and parsing job statuses, with detailed error handling for unexpected or failed responses.
* [`cav/subclient_vmware_jobs.go`](diffhunk://#diff-e78faa441b99bcbde097a5c7beca05d43f0bf8834b5241d023aced6c0c790a1fR123-R139): Improved VMware job parsing logic to validate job IDs as UUIDs and handle errors when job statuses or IDs fail to parse correctly. [[1]](diffhunk://#diff-e78faa441b99bcbde097a5c7beca05d43f0bf8834b5241d023aced6c0c790a1fR123-R139) [[2]](diffhunk://#diff-e78faa441b99bcbde097a5c7beca05d43f0bf8834b5241d023aced6c0c790a1fL141-R153)

### Middleware functionality:

* [`cav/endpoint.go`](diffhunk://#diff-e4bf7e8aa1fa3f9decec7af9042f5228b41e79c8fcf2ea9c25987e6a86bcf981R84-R88): Introduced `RequestMiddlewares` to allow custom modifications to HTTP requests before they are sent, such as adding headers or query parameters.
* [`cav/client_request.go`](diffhunk://#diff-639a5e20a07700b717644445bc4007774d9eaa19827c7f91ae5950124b9b0b3fR50-R57): Integrated middleware handling into the `NewRequest` method, ensuring middleware functions are applied when defined for an endpoint.

### Testing improvements:

* [`cav/subclient_cerberus_jobs_test.go`](diffhunk://#diff-9170a82e46909a14b3d97a4149ce53192884e3ad825479bd52550000f2b2b858R1-R198): Added comprehensive unit tests for Cerberus job parsing and status handling, covering edge cases like empty responses, unknown statuses, and error scenarios.
* [`cav/subclient_vmware_jobs_test.go`](diffhunk://#diff-44e165e6f6b1f03951b7a16f7f5ab1f69d574b0338a11ed437ce0e4734f5cce5R1-R204): Added unit tests for VMware job parsing, including validation of job IDs, handling of malformed headers, and error responses.

### Removal of unused dependencies:

* [`go.mod`](diffhunk://#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L8): Removed the `github.com/google/uuid` dependency, likely due to the introduction of a custom UUID validation mechanism in `common-go/validators`.

### Updates to mock clients:

* [`cav/mock/client.go`](diffhunk://#diff-0df7fc81021d1345eadfdafec1e169678d20e3f8161ef2c7c2e449226952fe08R72-R80): Modified mock behavior for `Cerberus` and `VMware` subclients, including handling specific HTTP status codes and registering mock response functions for endpoints. [[1]](diffhunk://#diff-0df7fc81021d1345eadfdafec1e169678d20e3f8161ef2c7c2e449226952fe08R72-R80) [[2]](diffhunk://#diff-0df7fc81021d1345eadfdafec1e169678d20e3f8161ef2c7c2e449226952fe08R89-R95) [[3]](diffhunk://#diff-0df7fc81021d1345eadfdafec1e169678d20e3f8161ef2c7c2e449226952fe08L100-R116)

### CODE REVIEW

File: cav/subclient_cerberus_jobs.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+	// ! This is untestable because resp.Bytes() is indefinable in the mock.
+	if resp.StatusCode() == http.StatusCreated {
+		jobCreated := &cerberusJobCreatedAPIResponse{}
+		if err := json.Unmarshal(resp.Bytes(), jobCreated); err == nil {
+			// Continue only if the unmarshalling was successful.
+			return &Job{
+				ID:          jobCreated.ID,
+				Description: jobCreated.Message,
+			}, nil
+		}
```
Comment (dmicheneau) [inline cav/subclient_cerberus_jobs.go:128]:
```suggestion
        if err := json.Unmarshal(resp.Bytes(), jobCreated); err == nil {
			// Continue only if the unmarshalling was successful.
			return &Job{
				ID:          jobCreated.ID,
				Description: jobCreated.Message,
			}, nil
		}
        // need to check if an error is return
        if err != nil {
		    return nil, errors.New("failed to unmarshall response: " + err.Error())
	    }
```

**Review by dmicheneau — APPROVED** [review]
